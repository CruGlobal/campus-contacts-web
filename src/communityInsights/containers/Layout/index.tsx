import React from 'react';
import styled from '@emotion/styled';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import Navigation from '../Navigation';
import PersonalStepsPage from '../PersonalStepsPage';
import StepsOfFaithPage from '../StepsOfFaithPage';
import InteractionsPage from '../InteractionsPage';
import ChallengesPage from '../ChallengesPage';

const LayoutWrapper = styled.div`
    display: flex;
    color: ${({ theme }) => theme.colors.primary};
    font-family: ${({ theme }) => theme.font.family};
    font-style: normal;
    font-weight: normal;
    font-size: ${({ theme }) => theme.font.size};
    line-height: 20px;
    flex-direction: column;
    background-color: #f2f2f2;
`;

const Content = styled.div`
    width: 100%;
`;

const Layout = () => {
    return (
        <LayoutWrapper>
            <Router>
                <Navigation />
                <Content>
                    <Route
                        path="/ministries/:id/insights/personal"
                        component={PersonalStepsPage}
                    />
                    <Route
                        path="/ministries/:id/insights/steps"
                        component={StepsOfFaithPage}
                    />
                    <Route
                        path="/ministries/:id/insights/interactions"
                        component={InteractionsPage}
                    />
                    <Route
                        path="/ministries/:id/insights/challenges"
                        component={ChallengesPage}
                    />
                </Content>
            </Router>
        </LayoutWrapper>
    );
};

export default Layout;
