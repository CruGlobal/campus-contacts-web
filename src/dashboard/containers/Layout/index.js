// Vendors
import React from 'react';
import styled from '@emotion/styled';
import { BrowserRouter as Router, Route } from 'react-router-dom';
// Projects
import Navigation from '../Navigation';
import PersonalStepsPage from '../PersonalStepsPage';
import StepsOfFaithPage from '../StepsOfFaithPage';

const LayoutWrapper = styled.div`
    display: flex;
    color: ${props => props.theme.colors.primary};
    font-family: ${props => props.theme.font.family};
    font-style: normal;
    font-weight: normal;
    font-size: ${props => props.theme.font.size};
    line-height: 20px;
    flex-direction: column;
    background-color: #f2f2f2;
`;

const Content = styled.div`
    width: 100%;
`;

const Layout = () => {
    return (
        <LayoutWrapper>
            <Router>
                <Navigation />
                <Content>
                    <Route
                        path="/ministries/:id/insights/personal"
                        component={PersonalStepsPage}
                    />
                    <Route
                        path="/ministries/:id/insights/steps"
                        component={StepsOfFaithPage}
                    />
                    <Route
                        path="/ministries/:id/insights/interactions"
                        component={StepsOfFaithPage}
                    />
                    <Route
                        path="/ministries/:id/insights/challenges"
                        component={StepsOfFaithPage}
                    />
                </Content>
            </Router>
        </LayoutWrapper>
    );
};

export default Layout;

Layout.propTypes = {};
