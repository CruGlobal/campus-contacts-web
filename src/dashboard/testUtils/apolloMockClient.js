import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
import { buildClientSchema } from 'graphql/utilities';
import { createMockerFromIntrospection } from 'fraql/mock';
import gql from 'fraql';

import introspectionQuery from '../../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument, options) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
