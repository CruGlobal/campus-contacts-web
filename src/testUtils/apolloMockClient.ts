import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
// @ts-ignore
import { buildClientSchema } from 'graphql/utilities';
// @ts-ignore
import { createMockerFromIntrospection } from 'fraql/mock';
// @ts-ignore
import gql from 'fraql';

import introspectionQuery from '../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument: any, options: any) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
