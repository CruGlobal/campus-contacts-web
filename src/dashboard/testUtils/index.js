import React, { ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';

import { createApolloMockClient } from './apolloMockClient';
import defaultTheme from '../defaultTheme';
import { ThemeProvider } from 'emotion-theming';
import AppContext from '../appContext';

export const renderWithContext = (component, { mocks } = {}, context = {}) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={context}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });

    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
