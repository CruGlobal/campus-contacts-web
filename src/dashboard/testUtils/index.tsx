import React, { ReactNode } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';

import defaultTheme from '../defaultTheme';
import AppContext from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

export const renderWithContext = (
    component: any,
    { mocks: mocks = {}, appContext = {} } = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    // @ts-ignore
    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
