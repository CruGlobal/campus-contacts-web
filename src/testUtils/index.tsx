import React, { ReactNode, ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import { IMocks } from 'graphql-tools';

import defaultTheme from '../defaultTheme';
import {
    AppContext,
    AppContextValue,
    appContextDefaultValue,
} from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

interface RenderWithContextParams {
    mocks?: IMocks;
    appContext?: AppContextValue;
}

export const renderWithContext = (
    component: ReactElement,
    {
        mocks: mocks = {},
        appContext = appContextDefaultValue,
    }: RenderWithContextParams = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children?: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
