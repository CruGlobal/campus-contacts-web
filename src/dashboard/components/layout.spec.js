import React from 'react';
import Layout from './layout';

import { renderWithContext } from '../testUtils';
import { ThemeProvider } from 'emotion-theming';

describe('<Layout />', () => {
    it('should render properly', async () => {
        const theme = {
            colors: {
                primary: '#505256',
            },
        };

        renderWithContext(
            <ThemeProvider theme={theme}>
                <Layout orgId={1} />
            </ThemeProvider>,
            { mocks: {} },
        ).snapshot();
    });
});
