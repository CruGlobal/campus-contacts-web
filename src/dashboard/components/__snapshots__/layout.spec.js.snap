// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render properly 1`] = `
<div>
  <div
    class="css-11tk6jz"
  >
    <div
      class="css-0"
    >
      Loading...
    </div>
    <div
      class="css-0"
    />
  </div>
</div>
`;
